lib/brains/models.py

```diff
@@ -24,12 +29,41 @@ def get_uri(self):
         return reverse('braintree:mozilla:buyer', kwargs={'pk': self.pk})
 
 
+@receiver(Buyer.close_signal, sender=Buyer)
+def close(signal, *args, **kw):
+    buyer = kw['buyer']
+    try:
+        braintree_buyer = BraintreeBuyer.objects.get(buyer=buyer)
+    except BraintreeBuyer.DoesNotExist:
+        # The buyer might not have used braintree. If that's the
+        # case continue.
+        log.info('No braintree buyer found for buyer: {}'
+                 .format(buyer.pk))
+        return
+
+    for paymethod in braintree_buyer.paymethods.all():
+
+        # Find and clear out all subscriptions.
+        for subscription in paymethod.subscriptions.filter(active=True):
+            subscription.braintree_cancel()
```

Comment:
I guess this will send an email, right? If someone is deleting their account will they want a bunch of emails for each of their cancelled subscriptions?